SeaPublicWebsite.Data/SeaDbContext.cs

```diff
@@ -41,4 +43,31 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             .HasForeignKey<PropertyData>("EditedDataId")
             .OnDelete(DeleteBehavior.Cascade);
     }
+
+    private void SetupPrimaryKeys(ModelBuilder modelBuilder)
```

Comment:
I think it will probably be more maintainable to have functions for each table, rather than for each thing that applies across all tables.
You could also move this into one of them:
```
modelBuilder.Entity<PropertyData>()
            .HasIndex(p => p.Reference)
            .IsUnique();
```